File: server_script.r
# Set up a socket client
sock <- socketConnection(host = "localhost", port = 9999, server = TRUE)
open(sock, "w")

# Send a message to the server
message <- "Hello world! This message is from the R server."
cat("Sending message:", message, "\n")
writeLines(message, sock)

# Clean up
close(sock)

File: server_script.py
import socket

# Set up a socket server
server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
server_socket.bind(('localhost', 9999))
server_socket.listen(1)

# Wait for a client to connect
print("Waiting for client to connect...")
client_socket, client_address = server_socket.accept()
print("Client connected:", client_address)

# Send a message to the client
message = "hello world"
print("Sending message:", message)
client_socket.sendall(message.encode())

# Clean up
client_socket.close()
server_socket.close()
